pkg/alertmanager/alertmanager_template.go

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package alertmanager
+
+import (
+	"encoding/json"
+	tmplhtml "html/template"
+	"net/url"
+	tmpltext "text/template"
+
+	"github.com/prometheus/alertmanager/template"
+)
+
+type grafanaExploreRange struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+type grafanaExploreQuery struct {
+	Datasource string `json:"datasource"`
+	Expr       string `json:"expr"`
+	Instant    bool   `json:"instant"`
+	Range      bool   `json:"range"`
+	RefID      string `json:"refId"`
+}
+
+type grafanaExploreParams struct {
+	Datasource string                `json:"datasource"`
+	Range      grafanaExploreRange   `json:"range"`
+	Queries    []grafanaExploreQuery `json:"queries"`
+}
+
+// grafanaURL is a template helper function to generate Grafana explore URL in the alert template.
+func grafanaURL(grafanaURL, datasource, from, to, expr string) (string, error) {
```

Comment:
[nit]
```suggestion
func grafanaExploreURL(grafanaURL, datasource, from, to, expr string) (string, error) {
```

Reply:
fixed in https://github.com/grafana/mimir/pull/3849/commits/e7abe5cb4c9c324dba89ff53ad3e7f636143cf31